### ISSUE TITLE
Kubernetes filter with custom logging path

### ISSUE THREAD

**Issue by ustndagsemih** [fluent/fluent-bit#7499]
Hello,

I need to write the logs to a file at a custom path like /logs/application.log for some reason. 

I know that Kubernetes filter needs the correct file name to populate Kubernetes metadata informations. Is there any other way to populate these information without needing to default path and file name format?

Basically instead of /var/log/containers/*.log, I want fluent-bit to read from /logs/*.log and be able to get metadata information from Kubernetes using kubernetes filter.

Thanks.

**Comment by ustndagsemih** [comment]
Thanks for the comment. I am aware of the naming format but wanted to ask if there is an alternative.

For the format, I think pod id is not relevant, right? Because with a default environment, pods are writing their logs with the format of **pod_name**-**namespace_name**-**container_name**-**docker_id**

I can get all the informations except **docker_id** with kubernetes downward API. If that is not necessary, I think i can make the application write their logs to custom path with the correct naming format.

**Comment by patrick-stephens** [comment]
I suspect that should be container ID, Docker was removed from K8S, so that is also available. Everything you need should be available via the downward API so that's an easy way to do it.

Make sure to set the kube tag prefix though.

**Comment by ustndagsemih** [comment]
Correct, it is container ID. However this information is not accessible through downward API.

https://kubernetes.io/docs/concepts/workloads/pods/downward-api/

I can only get UID but it is different from container ID. UID is something like _92b1f0c5-5f33-4df9-a5c6-cb2b52975f96_ so it does not fit the format. 

Normally container ID is like _0e4701597647f40a7d2a0a57de19791fb2b6619fda258ecf351e59a9b5ccc026_

Since Kubernetes does not provide this value, how should I proceed? Are there anything you can recommend?

**Comment by ustndagsemih** [comment]
I discarded container ID and changed regex parser to not include contaider ID. It works now. Thanks!

**Comment by zhengqz** [comment]
filter kubernetes
can support  pod  uid: 1d99a013-61a7-48bf-8dc6-fd5fb5a68f0f
deployment use empty dir volumeMount, and  write app log in this dir, which will mount bind  /var/log/<pod-uid>/volumes/kubernetes.io~empty-dir/log-dir/info.log

tag kube.<pod-uid>
